examples/react-app/src/App.tsx

```diff
@@ -638,15 +635,15 @@ class App extends React.Component<any, any> {
         throw new Error(`Missing chain data for chainId: ${chainId}`);
       }
 
-      // TODO: check if valid
-      const valid = true;
+      const messageHash = sha256(makeSignBytes(signDoc));
+      const valid = await verifyCosmosSignature(address, result.signature, messageHash);
 
       // format displayed result
       const formattedResult = {
         method: "cosmos_signDirect",
         address,
         valid,
-        result: result.signature.signature,
+        result: result.signature,
```

Comment:
Per the [spec](https://docs.walletconnect.org/v/2.0/json-rpc/cosmos#returns-1) the `signature` property is a string.